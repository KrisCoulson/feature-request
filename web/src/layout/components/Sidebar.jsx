import React from 'react';
import axios from 'axios';

const createNewPost = values => {
  axios.post('http://localhost:3001/post', {
    title: values.title,
    description: values.description
  });
};
const getPosts = values => {
  axios.get('http://localhost:3001/post').then(res => console.log(res.data))
};

class Sidebar extends React.Component {
  state = {
    title: '',
    description: ''
  };

  render() {
    return (
      <div
        style={{
          width: '400px',
          display: 'flex',
          flexDirection: 'column'
        }}>
        <label>Title</label>
        <input
          placeholder="Title"
          type="text"
          value={this.state.title}
          onChange={e => this.setState({ title: e.target.value })}
        />
        <label>Description</label>
        <textarea
          placeholder="Description"
          value={this.state.description}
          onChange={e => this.setState({ description: e.target.value })}
        />
        <button onClick={() => createNewPost(this.state)}>New Post</button>
        <button onClick={() => getPosts()}>Get Posts</button>
      </div>
    );
  }
}

export default Sidebar;
