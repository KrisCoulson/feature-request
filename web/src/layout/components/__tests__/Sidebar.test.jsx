import React from 'react';
import Sidebar from '../Sidebar';

describe('Sidebar Component', () => {
  it('should render', () => {
    expect(<Sidebar />).toMatchSnapshot();
  });
})
