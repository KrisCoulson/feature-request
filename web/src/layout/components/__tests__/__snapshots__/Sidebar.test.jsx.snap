// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Component should render 1`] = `<Sidebar />`;
